src/ui/common/src/components/integrations/dialogs/IntegrationFileUploadField.tsx

```diff
@@ -65,26 +70,53 @@ export const IntegrationFileUploadField: React.FC<
     e.stopPropagation();
   };
 
-  const handleDrop = (e) => {
-    e.preventDefault();
-    e.stopPropagation();
+  // const handleDrop = (e) => {
+  //   e.preventDefault();
+  //   e.stopPropagation();
 
-    setDragging(false);
+  //   setDragging(false);
 
-    const { files } = e.dataTransfer;
-    if (files && files.length) {
-      onFiles(files);
-    }
-  };
+  //   const { files } = e.dataTransfer;
+  //   if (files && files.length) {
+  //     onFiles(files);
+  //   }
+  // };
+
+  const handleDrop = useCallback(
+    (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      setDragging(false);
+
+      const { files } = e.dataTransfer;
+      if (files && files.length) {
+        onFiles(files);
+      }
+    },
+    [onFiles]
+  );
 
   useEffect(() => {
-    if (drop.current) {
-      drop.current.addEventListener('dragenter', handleDragEnter);
-      drop.current.addEventListener('dragleave', handleDragLeave);
-      drop.current.addEventListener('dragover', handleDragOver);
-      drop.current.addEventListener('drop', handleDrop);
+    const current = drop?.current;
+
+    if (current) {
+      current.addEventListener('dragenter', handleDragEnter);
+      current.addEventListener('dragleave', handleDragLeave);
+      current.addEventListener('dragover', handleDragOver);
+      current.addEventListener('drop', handleDrop);
     }
-  }, []);
+
+    // clean up event listeners
+    return () => {
```

Comment:
whenever adding event listeners, we should be sure to remove them as well. I noticed that they weren't being removed, so I've added this return statement here.